hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/snapshot/SnapshotDiffManager.java

```diff
@@ -1217,6 +1223,23 @@ Set<String> getDeltaFiles(OmSnapshot fromSnapshot,
             toSnapshot.getSnapshotTableKey()));
   }
 
+  private Set<String> getDiffFiles(OmSnapshot fromSnapshot, OmSnapshot toSnapshot, List<String> tablesToLookUp) {
+    Set<String> diffFiles;
+    try {
+      Map<Object, String> fromSnapshotFiles = getSSTFileMapForSnapshot(fromSnapshot, tablesToLookUp);
+      Map<Object, String> toSnapshotFiles = getSSTFileMapForSnapshot(toSnapshot, tablesToLookUp);
+      diffFiles = Stream.concat(fromSnapshotFiles.entrySet().stream(),
+              toSnapshotFiles.entrySet().stream().filter(e -> !fromSnapshotFiles.containsKey(e.getKey())))
```

Comment:
This logic seems to be wrong. You would only get diff files which are present in toSnapshot
Consider
FromSnapshot has => 1.sst,2.sst, 3.sst
toSnapshot has => 2.sst, 3.sst, 5.sst
Diff should be 1.sst & 5.sst. But this logic will return only 5.sst

Reply:
```suggestion
              toSnapshotFiles.entrySet().stream().filter(e -> !(fromSnapshotFiles.containsKey(e.getKey()) && toSnapshot.containsKey(e.getKey()))))
```

Reply:
done.